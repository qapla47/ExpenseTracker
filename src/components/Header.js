import React from "react";
import ReactDOM from "react-dom";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expense Tracker</h1>
    <NavLink exact to="/" className="nav-link" activeClassName="is-active">
      Dashboard
    </NavLink>{" "}
    <NavLink to="/create" className="nav-link" activeClassName="is-active">
      Create an Expense
    </NavLink>{" "}

  </header>
);

// const ExpenseDashboardPage = () => (
//   <div>This is from my dashboard component</div>
// );


export default Header;
