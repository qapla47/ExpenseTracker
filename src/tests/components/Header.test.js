import React from "react";
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from "../../components/Header";
// test shallow to check only what is being rendered, rather than everything that's being rendered

test("should render header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

// snapshots allow us to track changes to data over time

