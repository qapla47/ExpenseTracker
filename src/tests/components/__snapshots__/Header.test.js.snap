// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expense Tracker
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    className="nav-link"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
   
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    className="nav-link"
    to="/create"
  >
    Create an Expense
  </NavLink>
   
</header>
`;
